src/ahocorasick/AhoCorasickPlus.cpp

```diff
@@ -34,21 +38,22 @@ AhoCorasickPlus::~AhoCorasickPlus ()
     delete m_acText;
 }
 
+
 AhoCorasickPlus::EnumReturnStatus AhoCorasickPlus::addPattern
-    (const std::string &pattern, PatternId id)
+    (const char* pattern, size_t len, PatternId id, bool copy)
```

Comment:
`string_view` instead of a pointer and a length?

Reply:
Probably, but doesn't matter too much if we go with this for now since it's not user-facing API and also if we want to port this into a Zeek 3.0.x, we will need a patch that's C++11-only anyway.

Reply:
Pushed a commit to change to `string_view` on top of this just so we have the original to cherry-pick if we want.